addons/binding/org.openhab.binding.hs110/src/main/java/org/openhab/binding/hs110/internal/HS110.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.hs110.internal;
+
+import static org.eclipse.smarthome.core.library.types.OnOffType.ON;
+import static org.openhab.binding.hs110.internal.Util.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigDecimal;
+import java.net.ConnectException;
+import java.net.InetAddress;
+import java.net.Socket;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ *
+ * The {@link HS110} acts as a protocol parser and interface to the physical device
+ *
+ * @author Christian Fischer - Initial contribution
+ */
+public class HS110 {
+
+    private Logger logger = LoggerFactory.getLogger(HS110.class);
+
+    private static final int HS100_PORT = 9999;
+
+    public enum Command {
+        SWITCH_ON("{\"system\":{\"set_relay_state\":{\"state\":1}}}}"),
+        SWITCH_OFF("{\"system\":{\"set_relay_state\":{\"state\":0}}}}"),
+        SYSINFO("{\"system\":{\"get_sysinfo\":null}}"),
+        ENERGY("{\"emeter\":{\"get_realtime\":null}}");
+
+        public String value;
```

Comment:
better not expose the inner fields of an enum like, I would prefer to have it private with a getter. Less proper but acceptable alternative would be to make the field final so you do no accidentally overwrite it somewhere